### PR TITLE
React 18: Custom test fixes for React 18/RTL update

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

This PR is the part of a chain of PRs upgrading Perseus to React 18.

  1. #1400
  2. #1401
  3. #1402
  4. This PR --> #1403

A few tests needed larger changes than simply wrapping the actions in `act()` (see #1402). This PR covers the remaining tests that were failing for various reasons. In some cases the tests were largely re-written (sometimes for better test clarity while re-working them to pass). 

In the case of the Zoomable tests, we were waiting for the text to be updated in the element, but the styling changes had not yet settled as the updates caused multiple `setState()` calls to be made which all needed to be called and then resolved by a subsequent render (thus the new `waitForStyle()` helper).

In the Graded Group test update, I'm not sure how it worked as written. In the test that was fixed, we were testing mobile and in that case the "Check" button reads "Try again" on the second and subsequent attempts. 

In the Group tests, I needed to wrap some actions in `act()` as usual, but I also took the opportunity to change the element finding logic away from the `getAll...()` functions to depend more on the aria labels. This makes these tests read more clearly (instead of accessing item at index `[0]` we get the textbox with the string `"value rounded to the nearest ten"`).

Lastly, there was one outdated snapshot due to a test file being renamed.

Issue: LEMS-1802

## Test plan:

`yarn test` all tests now pass!

### Before 

```sh
$ yarn test 
...
Test Suites: 3 failed, 1 skipped, 185 passed, 188 of 189 total
Tests:       6 failed, 14 skipped, 2567 passed, 2587 total
Snapshots:   4 failed, 1 file obsolete, 155 passed, 159 total
Time:        22.175 s, estimated 27 s
Ran all test suites.
```

### After 

```sh
$ yarn test 
...
Test Suites: 1 skipped, 188 passed, 188 of 189 total
Tests:       14 skipped, 2573 passed, 2587 total
Snapshots:   159 passed, 159 total
Time:        26.765 s
Ran all test suites.
```

### CODE REVIEW

File: packages/perseus/src/components/__tests__/zoomable.test.tsx [hunk 1]
```diff
@@ -214,8 +214,10 @@ describe("Zoomable", () => {
         // Arrange
         // Simulate window resize event
         const resizeWindowTo = (width: number, height: number) => {
-            const resizeEvent = document.createEvent("Event");
-            resizeEvent.initEvent("resize", true, true);
```
Comment (jeremywiebe) [inline packages/perseus/src/components/__tests__/zoomable.test.tsx:218]: ESLint was complaining that `initEvent` is deprecated. Given this is test code and the test still passes, I considered it a safe update.

File: packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap
Comment (jeremywiebe) [inline packages/perseus/src/widgets/__tests__/__snapshots__/interactive-graph.test.ts.snap:1]: The file under test was renamed at some point earlier to `.tsx` and so this snapshot is unused today.

File: packages/perseus-editor/src/__tests__/editor.test.tsx [hunk 2]
```diff
@@ -123,7 +123,6 @@ describe("Editor", () => {
         await userEvent.clear(captionInput);
         await userEvent.type(captionInput, "A picture of kittens");
         await userEvent.tab(); // blurring the input triggers onChange to be called
-        jest.runOnlyPendingTimers();
```
Comment (jeremywiebe) [inline packages/perseus-editor/src/__tests__/editor.test.tsx:126]: Tests still pass without this, so I'm removing it.

**Review by jeresig — APPROVED** [review]
Glad you were able to find some good fixes here, seems tricky!

**Pull request merged**